src/System.Sorting/tests/SortTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Linq;
+using Xunit;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Sorting.Tests
+{
+    public class SortTests
+    {
+        [Fact]
+        public static void TestSort()
+        {
+            IComparer<int> icomparer = new IntegerComparer();
+
+            TestSortHelper<int>(new int[] { }, 0, 0, icomparer);
+            TestSortHelper<int>(new int[] { 5 }, 0, 1, icomparer);
+            TestSortHelper<int>(new int[] { 5, 2 }, 0, 2, icomparer);
+
+            TestSortHelper<int>(new int[] { 5, 2, 9, 8, 4, 3, 2, 4, 6 }, 0, 9, icomparer);
+            TestSortHelper<int>(new int[] { 5, 2, 9, 8, 4, 3, 2, 4, 6 }, 3, 4, icomparer);
+            TestSortHelper<int>(new int[] { 5, 2, 9, 8, 4, 3, 2, 4, 6 }, 3, 6, icomparer);
+
+            IComparer<string> scomparer = new StringComparer();
+            TestSortHelper<String>(new string[] { }, 0, 0, scomparer);
+            TestSortHelper<String>(new string[] { "5" }, 0, 1, scomparer);
+            TestSortHelper<String>(new string[] { "5", "2" }, 0, 2, scomparer);
+
+            TestSortHelper<String>(new string[] { "5", "2", null, "8", "4", "3", "2", "4", "6" }, 0, 9, scomparer);
+            TestSortHelper<String>(new string[] { "5", "2", null, "8", "4", "3", "2", "4", "6" }, 3, 4, scomparer);
+            TestSortHelper<String>(new string[] { "5", "2", null, "8", "4", "3", "2", "4", "6" }, 3, 6, scomparer);
+        }
+
+
+        private static void TestSortHelper<T>(T[] array, int index, int length, IComparer<T> comparer)
+        {
+
+
+            T[] control = SimpleSort<T>(array, index, length, comparer);
+
+            //TODO - Can use Reflection to get the derivedTypes of Sorting and invoking the methods. 
+            //It's not so trivial because we are using Reflection with Generic <T>. 
+            //TypeInfo[] mySortTestAlgorithms = FindAllDerivedTypesOfSorting();
+
+            Sorting[] sortAlgos = new Sorting[] { new IntroSorting(), new InsertionSorting() };
+            foreach(var algo in sortAlgos)
+            {
+                T[] spawn = (T[])(array.Clone());
+                // get instance from algo
+                algo.Sort<T>(spawn, index, length, comparer);
+                Assert.True(ArraysAreEqual<T>((T[])spawn, control, comparer));
+            }
+
+         }
+
+        private static T[] SimpleSort<T>(T[] a, int index, int length, IComparer<T> comparer)
```

Comment:
Instead of doing this, it might be better to just walk the array and test if each element is greater than or equal to the preceding element.